opensearch-operator/pkg/builders/cluster.go

```diff
@@ -972,3 +984,67 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *prometheus.ServiceMonitor {
+
+	labels := map[string]string{
+		ClusterLabel: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := prometheus.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+
+	scrape := cr.Spec.General.Monitoring.ScrapInterval
+	if scrape == "" {
+		scrape = "30s"
+	}
+
+	monitorUser := cr.Spec.General.Monitoring.MonitoringUser
+	if monitorUser == "" {
+		monitorUser = "admin"
+	}
+
+	return &prometheus.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-monitor",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: prometheus.ServiceMonitorSpec{
+			JobLabel: cr.Name + "-monitor",
+			TargetLabels: []string{
+				cr.Name,
+			},
+			PodTargetLabels: []string{
+				cr.Name,
+			},
+			Endpoints: []prometheus.Endpoint{
+				{Port: "9200",
+					TargetPort:      nil,
+					Path:            "/_prometheus/metrics",
+					Interval:        scrape,
+					TLSConfig:       nil,
+					BearerTokenFile: "",
+					HonorLabels:     false,
+					BasicAuth: &prometheus.BasicAuth{
+						Username: corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + monitorUser + "-password"},
+							Key:                  monitorUser,
```

Comment:
This is a logic error. You cannot use the user as the key. Did you intend to use `username` as key?

Reply:
yeah, my bad, fixed